src/common/cockpitpipe.c

```diff
@@ -925,14 +1010,25 @@ _cockpit_pipe_write (CockpitPipe *self,
       return;
     }
 
-  if (g_bytes_get_size (data) == 0)
+  size = g_bytes_get_size (data);
+  if (size == 0)
     {
       g_debug ("%s: ignoring zero byte data block", self->priv->name);
       return;
     }
 
+  before = self->priv->out_queued;
+  g_return_if_fail (G_MAXSIZE - size > self->priv->out_queued);
+  self->priv->out_queued += size;
   g_queue_push_tail (self->priv->out_queue, g_bytes_ref (data));
 
+  /*
+   * If we have two much data queued, and are controlling another flow
+   * tell it to stop sending data, each time we cross over the high bound.
+   */
+  if (before < QUEUE_PRESSURE && self->priv->out_queued >= QUEUE_PRESSURE)
```

Comment:
What do you think would happen if we immediately applied backpressure as soon as we had even 1 item in the queue, and didn't realise it until the queue was empty again?

ie: QUEUE_PRESSURE = 1